src/tls.c

```diff
@@ -1021,9 +1024,7 @@ static ssize_t connTLSSyncRead(connection *conn_, char *ptr, ssize_t size, long
     tls_connection *conn = (tls_connection *)conn_;
 
     setBlockingTimeout(conn, timeout);
-    ERR_clear_error();
-    int ret = SSL_read(conn->ssl, ptr, size);
-    ret = updateStateAfterSSLIO(conn, ret, 0);
+    int ret = connTLSRead(conn_, ptr, size);
```

Comment:
I think this will cause us to potentially keep updating the event in updateStateAfterSSLIO. Same for the  connTLSSyncReadLine. Maybe you want to create an internal function which will also get the update_event indication?

Reply:
Thanks, I reverted this change. Instead, we will call `updatePendingFlags` explicitly.